cishouseholds/derive.py

```diff
@@ -37,7 +38,48 @@ def substring_column(df: DataFrame, new_column_name, column_to_substr, start_pos
     Return
     ------
     df: pyspark.sql.DataFrame
+
     """
     df = df.withColumn(new_column_name, F.substring(column_to_substr, start_position, len_of_substr))
 
     return df
+
+
+def derive_ctpattern(df: DataFrame, spark_session):
+    """
+    Derive a new column containing string of pattern in
+    ["N only", "OR only", "S only", "OR+N", "OR+S", "N+S", "OR+N+S", NULL]
+    indicating which ct_* columns indicate a positive result.
+    From households_aggregate_processes.xlsx, derivation number 7.
+
+    Parameters
+    ----------
+    spark_session: pyspark.sql.SparkSession
+    df: pyspark.sql.DataFrame
+
+    Return
+    ------
+    df: pyspark.sql.DataFrame
+    """
+    indicator_list = ["indicator_ct_or", "indicator_ct_n", "indicator_ct_s"]
+
+    lookup_df = spark_session.createDataFrame(
+        data=[
+            (0, 0, 0, None),
+            (1, 0, 0, "OR only"),
+            (0, 1, 0, "N only"),
+            (0, 0, 1, "S only"),
+            (1, 1, 0, "OR+N"),
+            (1, 0, 1, "OR+S"),
+            (0, 1, 1, "N+S"),
+            (1, 1, 1, "OR+N+S"),
+        ],
+        schema=indicator_list + ["ctpattern"],
+    )
+
+    for ct_gene in ["or", "n", "s"]:
```

Comment:
Check - are these (eg. `ct_or`) the correct variable names?

Reply:
Also, are these variable nullable?

Reply:
Variable names not decided - give to function as variable.
Three variables are nullable - should behave as 0